velox/functions/lib/Re2Functions.cpp

```diff
@@ -300,6 +342,43 @@ class Re2SearchAndExtract final : public VectorFunction {
   }
 };
 
+class LikeConstantPattern final : public VectorFunction {
+ public:
+  explicit LikeConstantPattern(StringView pattern, char escapeChar)
+      : re_(toStringPiece(likePatternToRe2(pattern, escapeChar)), RE2::Quiet) {}
+
+  void apply(
+      const SelectivityVector& rows,
+      std::vector<VectorPtr>& args,
+      Expr* /* caller */,
+      EvalCtx* context,
+      VectorPtr* resultRef) const final {
+    VELOX_CHECK(args.size() == 2 || args.size() == 3);
+
+    // apply() will not be invoked if the selection is empty.
+    checkForBadPattern(re_);
```

Comment:
perhaps, move this check to the constructor; no need to check again and again on each call to apply()

Reply:
I checked for bad patterns in `apply` so an exception was thrown only if the expression was executed.